### ISSUE TITLE
bug(typescript): tsc 2.2 is unable to generate declarations when Mixins are used

### ISSUE THREAD

**Issue by Hotell** [wc-catalogue/blaze-elements#300]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => You can submit support request here, but we preffer to chat about it realtime at our [ngParty slack](https://ngparty.herokuapp.com/)
```

**Current behavior**

```
50 export default class Button extends Colored( Disabled( Component ) )<ButtonProps> {
                        ~~~~~~

packages/button/Button.tsx(50,22): error TS4093: 'extends' clause of exported class 'Button' refers to a type whose name cannot be referenced.
```

**Expected behavior**

No errors when using `declaration: true`

**Comment by Hotell** [comment]
related https://github.com/Microsoft/TypeScript/issues/14597

**Comment by Hotell** [comment]
Closed by #303
